src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ProvideCorrectArgumentsToFormattingMethods.cs

```diff
@@ -358,16 +364,17 @@ public StringFormatInfo(Compilation compilation)
                 var determineAdditionalStringFormattingMethodsAutomatically = context.Options.GetBoolOptionValue(EditorConfigOptionNames.TryDetermineAdditionalStringFormattingMethodsAutomatically,
                         Rule, context.Operation.Syntax.SyntaxTree, context.Compilation, defaultValue: false);
                 if (determineAdditionalStringFormattingMethodsAutomatically &&
-                    TryGetFormatInfo(method, out info) &&
+                    TryGetFormatInfoByParameterName(method, out info) &&
                     info.ExpectedStringFormatArgumentCount == -1)
                 {
                     return info;
                 }
 
+                _map.TryAdd(method, null);
```

Comment:
So now we are going to save all invoked methods, I don't think we want that

Reply:
It is either re-checking every invocation of the same method, meaning more CPU time or using more memory.